From 9d2acad4f8d2e6a04b3b16e75978c47ca33fd221 Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Tue, 7 Jun 2022 13:13:11 +0200
Subject: [PATCH] mpc85xx: enable NAND support for all subtargets

In subtarget p2020, there wasn't enabled nand support, and because of
that there weren't available tools from mtd-utils package, which has
utilities for NAND flash memory even though reference board, which
is the only currently supported device in p2020 subtarget has NAND [1].

All subtargets in mpc85xx has already enabled nand support, let's do it
globally.

[1] https://www.nxp.com/design/qoriq-developer-resources/p2020-reference-design-board:P2020RDB

Patch was already submitted to the OpenWrt main repository:
https://github.com/openwrt/openwrt/pull/10005

Signed-off-by: Josef Schlehofer <pepe.schlehofer@gmail.com>
---
 target/linux/mpc85xx/Makefile        | 2 +-
 target/linux/mpc85xx/p1010/target.mk | 1 -
 target/linux/mpc85xx/p1020/target.mk | 1 -
 3 files changed, 1 insertion(+), 3 deletions(-)

diff --git a/target/linux/mpc85xx/Makefile b/target/linux/mpc85xx/Makefile
index 6bf096740e..398e578882 100644
--- a/target/linux/mpc85xx/Makefile
+++ b/target/linux/mpc85xx/Makefile
@@ -8,7 +8,7 @@ ARCH:=powerpc
 BOARD:=mpc85xx
 BOARDNAME:=Freescale MPC85xx
 CPU_TYPE:=8540
-FEATURES:=squashfs ramdisk
+FEATURES:=squashfs ramdisk nand
 SUBTARGETS:=p1010 p1020 p2020
 
 KERNEL_PATCHVER:=5.10
diff --git a/target/linux/mpc85xx/p1010/target.mk b/target/linux/mpc85xx/p1010/target.mk
index 12ed78ace1..019536ff1a 100644
--- a/target/linux/mpc85xx/p1010/target.mk
+++ b/target/linux/mpc85xx/p1010/target.mk
@@ -1,5 +1,4 @@
 BOARDNAME:=P1010
-FEATURES+=nand
 KERNELNAME:=simpleImage.tl-wdr4900-v1
 
 define Target/Description
diff --git a/target/linux/mpc85xx/p1020/target.mk b/target/linux/mpc85xx/p1020/target.mk
index a25409eebb..000fc64644 100644
--- a/target/linux/mpc85xx/p1020/target.mk
+++ b/target/linux/mpc85xx/p1020/target.mk
@@ -1,5 +1,4 @@
 BOARDNAME:=P1020
-FEATURES+=nand
 
 define Target/Description
 	Build firmware images for Freescale P1020 based boards.
-- 
2.34.1

